core/src/main/java/org/jdbi/v3/tweak/CollectorFactory.java

```diff
@@ -30,7 +29,7 @@
      * @param type the object type
      * @return {@code true}, if accepts, otherwise {@code false}
      */
-    boolean accepts(Class<?> type);
+    boolean accepts(ResolvedType type);
```

Comment:
Yeah, makes sense to try to avoid that.  However, consider what you can do with a raw `Type` -- all you can do is figure out what it actually is via `instanceof` and downcast and hope.  We could then only provide JDK `Type` subclasses e.g. `Class` and `ParameterizedType` and each plugin would have to figure out how to introspect it itself.

Whereas if we just pass the `ResolvedType` it's clear where you go from there.

Not sure which concern is greater here.  I don't see a particularly good solution either way :(